src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -216,153 +216,187 @@ public void updateReplicateStatus(String chainTaskId,
+     */
     @Retryable(value = {OptimisticLockingFailureException.class}, maxAttempts = 100)
     public Optional<TaskNotificationType> updateReplicateStatus(String chainTaskId,
                                                                 String walletAddress,
                                                                 ReplicateStatusUpdate statusUpdate) {
+        log.info("Replicate update request [status:{}, chainTaskId:{}, walletAddress:{}, details:{}]",
+                statusUpdate.getStatus(), chainTaskId, walletAddress, statusUpdate.getDetails());
 
-        ReplicateStatus newStatus = statusUpdate.getStatus();
-        ReplicateStatusModifier modifier = statusUpdate.getModifier();
-
-        ReplicateStatus currentStatus = null;
-        Replicate replicate;
-        Optional<ReplicatesList> replicates = getReplicatesList(chainTaskId);
-        if (replicates.isEmpty()) {
-            log.error(getReplicateStatusErrorLog("replicateList missing",chainTaskId, walletAddress, currentStatus, newStatus, modifier));
+        Optional<ReplicatesList> oReplicateList = getReplicatesList(chainTaskId);
+        if (oReplicateList.isEmpty() || oReplicateList.get().getReplicateOfWorker(walletAddress).isEmpty()) {
+            log.error("Cannot update replicate, could not get replicate [chainTaskId:{}, UpdateRequest:{}]",
+                    chainTaskId, statusUpdate);
             return Optional.empty();
         }
 
-        Optional<Replicate> optionalReplicate = replicates.get().getReplicateOfWorker(walletAddress);
-        if (optionalReplicate.isEmpty()) {
-            log.error(getReplicateStatusErrorLog("replicate missing", chainTaskId, walletAddress, currentStatus, newStatus, modifier));
-            return Optional.empty();
-        }
-        replicate = optionalReplicate.get();
-        currentStatus = replicate.getCurrentStatus();
+        ReplicatesList replicatesList = oReplicateList.get();
+        Replicate replicate = replicatesList.getReplicateOfWorker(walletAddress).get();
+        ReplicateStatus newStatus = statusUpdate.getStatus();
 
-        if (modifier.equals(ReplicateStatusModifier.WORKER) &&
-                !ReplicateWorkflow.getInstance().isValidTransition(currentStatus, newStatus)) {
-            log.error(getReplicateStatusErrorLog("bad workflow transition",chainTaskId, walletAddress, currentStatus, newStatus, modifier));
+        boolean isValidTransition = ReplicateWorkflow.getInstance()
+                .isValidTransition(replicate.getCurrentStatus(), newStatus);
+        if (!isValidTransition) {
+            log.error("Cannot update replicate, bad wokfow transition {}",
+                    getLogDetails(chainTaskId, replicate, statusUpdate));
             return Optional.empty();
         }
 
-        replicate = updateReplicateFields(chainTaskId, replicate, statusUpdate);
+        boolean canUpdate = true;
+
+        switch (newStatus) {
+            case CONTRIBUTE_FAILED:
```

Comment:
We don't want this status anymore

Reply:
As discussed, we can include CONTRIBUTE_FAILED and REVEAL_FAILED but there should be no update performed on the replicate.